mysql-test/suite/encryption/t/upgrade_crypt_data_v1.test

```diff
@@ -0,0 +1,128 @@
+call mtr.add_suppression("Failed to create or recover replication info repositories.");
+call mtr.add_suppression("db.opt file not found for test database. Using default Character set");
+call mtr.add_suppression("Skip re-populating collations and character sets tables in InnoDB read-only mode");
+call mtr.add_suppression("Skipped updating resource group metadata in InnoDB read only mode");
+call mtr.add_suppression("file './test/t1.ibd' is missing!");
+#TODO : This warning is caused by bug#
+call mtr.add_suppression("Table encryption flag is OFF in the data dictionary but the encryption flag in file ./test/t6.ibd is ON.");
+--enable_query_log
+
+--echo # Set different paths for --datadir
+let $MYSQLD_DATADIR1 = $MYSQL_TMP_DIR/data_8016_crypt_v1/data;
+
+--copy_file $MYSQLTEST_VARDIR/std_data/data_8016_crypt_v1.zip $MYSQL_TMP_DIR/data_8016_crypt_v1.zip
+
+--echo # Check that the file exists in the working folder.
+--file_exists $MYSQL_TMP_DIR/data_8016_crypt_v1.zip
+
+--echo # Unzip the zip file.
+--exec unzip -qo $MYSQL_TMP_DIR/data_8016_crypt_v1.zip -d $MYSQL_TMP_DIR
+
+--let $MYSQLD_DATADIR=`SELECT @@datadir`
+
+--echo # Stop DB server which was created by MTR default
+--source include/shutdown_mysqld.inc
+
+let $MYSQLD_LOG= $MYSQLTEST_VARDIR/log/update_crypt_data_v1.log;
+
+--echo # Restart the server on 8.0.16 directory with tables with crypt data v1. The tables were created as:
+--echo # CREATE TABLE t1 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255)) ENCRYPTION='Y';
+--echo # CREATE TABLE t2 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255)) ENCRYPTION='KEYRING';
+--echo # CREATE TABLE t3 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255));
+--echo # CREATE TABLE t4 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255)) ENCRYPTION_KEY_ID=4;
+--echo # CREATE TABLE t5 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255)) ENCRYPTION='N';
+--echo # CREATE TABLE t6 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255)) ENCRYPTION_KEY_ID=4;
+--echo # Table t1 is the only table because of which the upgrade is not possible. This is because it is only partially re-encrypted
+--echo # from master key encryption to online keyring encryption.
+
+--error 1
+--exec $MYSQLD_CMD --log-error=$MYSQLD_LOG --loose-skip-log-bin --skip-log-slave-updates --datadir=$MYSQLD_DATADIR1 --keyring_file_data=$MYSQL_TMP_DIR/data_8016_crypt_v1/mysecret_keyring $KEYRING_PLUGIN_OPT $KEYRING_PLUGIN_EARLY_LOAD
+
+--let ABORT_ON=NOT_FOUND
+--let SEARCH_FILE=$MYSQLD_LOG
+--let SEARCH_PATTERN=Upgrade is not possible as there are tables for which re-encryption from master key encryption to online keyring encryption was not finished\. Please either finish off the re-encryption or decrypt the tables\. This feature is experimental in the version you are upgrading from\.
+--source include/search_pattern_in_file.inc
+
+--echo # Remove the problematic tablespace and re-try the upgrade. It should pass.
+--remove_file $MYSQLD_DATADIR1/test/t1.ibd
```

Comment:
wouldn't there be stale DD entry, and InnoDB should abort startup complaining that IBD file is missing?

Reply:
It will complain t1 cannot be find (warning) but will not abort the startup.